pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -214,276 +210,275 @@ let
+
+        inherit pname version;
+
+        compiler = ghc;
+
+
+        getBuildInputs = {
+          inherit propagatedBuildInputs otherBuildInputs allPkgconfigDepends;
+          haskellBuildInputs = isHaskellPartition.right;
+          systemBuildInputs = isHaskellPartition.wrong;
+        };
+
+        env = buildHaskellPackages.shellFor {
+          packages = p: [ drv ];
+        };
+
+        isHaskellLibrary = isLibrary;
+
+        # TODO: ask why the split outputs are configurable at all?
+        # TODO: include tests for split if possible
+        # Given the haskell package, returns
+        # the directory containing the haddock documentation.
+        # `null' if no haddock documentation was built.
+        # TODO: fetch the self from the fixpoint instead
+        haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
+
+      };
+
+      meta = { inherit homepage license platforms; }
+             // optionalAttrs broken               { inherit broken; }
+             // optionalAttrs (description != "")  { inherit description; }
+             // optionalAttrs (maintainers != [])  { inherit maintainers; }
+             // optionalAttrs (hydraPlatforms != null) { inherit hydraPlatforms; }
+             ;
+
+    }
+    // optionalAttrs (preCompileBuildDriver != "")  { inherit preCompileBuildDriver; }
+    // optionalAttrs (postCompileBuildDriver != "") { inherit postCompileBuildDriver; }
+    // optionalAttrs (preUnpack != "")      { inherit preUnpack; }
+    // optionalAttrs (postUnpack != "")     { inherit postUnpack; }
+    // optionalAttrs (patches != [])        { inherit patches; }
+    // optionalAttrs (patchPhase != "")     { inherit patchPhase; }
+    // optionalAttrs (preConfigure != "")   { inherit preConfigure; }
+    // optionalAttrs (postConfigure != "")  { inherit postConfigure; }
+    // optionalAttrs (preBuild != "")       { inherit preBuild; }
+    // optionalAttrs (postBuild != "")      { inherit postBuild; }
+    // optionalAttrs (doBenchmark)          { inherit doBenchmark; }
+    // optionalAttrs (checkPhase != "")     { inherit checkPhase; }
+    // optionalAttrs (preCheck != "")       { inherit preCheck; }
+    // optionalAttrs (postCheck != "")      { inherit postCheck; }
+    // optionalAttrs (preInstall != "")     { inherit preInstall; }
+    // optionalAttrs (installPhase != "")   { inherit installPhase; }
+    // optionalAttrs (postInstall != "")    { inherit postInstall; }
+    // optionalAttrs (preFixup != "")       { inherit preFixup; }
+    // optionalAttrs (postFixup != "")      { inherit postFixup; }
+    // optionalAttrs (dontStrip)            { inherit dontStrip; }
+    // optionalAttrs (hardeningDisable != []) { inherit hardeningDisable; }
+    // optionalAttrs (stdenv.buildPlatform.libc == "glibc"){ LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive"; }
+    );
+in
+  drv // {
+    env = buildHaskellPackages.shellFor {
+      packages = p: [ drv ];
```

Comment:
Is this a duplicate of line 432? We should only have the `passthru` one.

Reply:
Oh yeah good point, will remove